var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -94,21 +95,24 @@ def install(self, spec, prefix):
             config_args.append('--enable-parallel4')
 
         CPPFLAGS.append("-I%s/include" % spec['hdf5'].prefix)
-        LDFLAGS.append("-L%s/lib"     % spec['hdf5'].prefix)
+        LDFLAGS.append("-L%s/lib" % spec['hdf5'].prefix)
+        for lib in os.listdir(join_path(spec['hdf5'].prefix, 'lib')):
+            if lib.endswith('.so'):
```

Comment:
```python
for lib in os.listdir(spec['hdf5'].prefix.lib):
    LIBS.append('-l{0}'.format(lib.replace('.{0}'.format(dso_suffix), '')))
```
ought to do the trick. Or just manually type them in if there aren't that many and you don't expect them to change. By the way, wasn't @alalazo working on a PR (#1682) that would handle this case nicely?